builtin/credential/approle/path_role.go

```diff
@@ -2355,9 +2375,28 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 		return nil, err
 	}
 
+	var numUses int
+	// Check whether or not num_uses is defined, otherwise fallback to secret_id_num_uses
+	if numUsesRaw, ok := data.GetOk("num_uses"); ok {
+		if numUses < 0 {
+			return logical.ErrorResponse("num_uses cannot be negative"), nil
+		}
+		numUses = numUsesRaw.(int)
+	} else {
+		numUses = role.SecretIDNumUses
```

Comment:
After some internal discussions, we have decided that we would like to bound the per SecretID num_uses and ttl by the corresponding values in the Role parameters. Like, if we set the num_uses for a secretID to 200 and the role. SecretIDNumUses is 100, the value for the role should be used. And the same for the ttl.

Reply:
Wouldn't this defeat the purpose of the feature request that was filed? Here it is specifically requested to separate the two and in particular, be able to override the role's configuration.
Without the possibility to override the role's configuration you would still have to create a separate role.

Reply:
This seems consistent with the existing rules for CIDR block restrictions...if the role value is set, then the per secret-id values must be a subset. Likewise, if the role value is _not_ set, then the per secret-id values are unbounded. I think this would still meet the original feature request, since the user can choose to set maximum values as part of the role config (or leave them unset and allow any value per-secret ID)

Reply:
> The override is still possible though, the only constraint is that the new secretID ttl/num_uses could not be more than the corresponding ones in the role config. So, this way, the Role should be configured with highest required values for those parameters, and this way, the secretIDs could be configured to have different values but in a defined range.

Okay this makes more sense to me. While I still would disagree with a couple of points, it looks like the original requestee has given his stamp of approval.
I'll think of a solution that would fit best for the situation.

Reply:
@hghaf099 @kent007 After a short consiledation, I came to the conclusion to simply lower bind the SecretID's ttl and num_uses to the Role's ttl and num_uses. This yields to a PR without any changes outside of the scope on the Role's configuration, possibly requiring users to change the approach they take when creating a Role.
For the best result possible I'll go for a change where validation is performed rather than implicitly overriding the SecretId's defined ttl/num_uses with the Role's ttl/num_uses.

Reply:
Hi @RemcoBuddelmeijer, I'm wondering what else remains to be done for this implementation?  It sounds like we just need a few additional checks on the values presented at secret-id creation time...I have some time available this month to finish it  up if need be

Reply:
Hey @kent007 I was unavailable the past days but I have the weekend planned to finish up the implementation!

Reply:
Intermediate update. My days have been packed far more than expected. Right now I am out of the country and expecting to be back next weekend.

Reply:
Updated the branch with the new requirements. Updated the upstream and merged main back into this branch. Will change changelog. Please feel free to review!